src/System.Private.CoreLib/shared/System/Globalization/CharUnicodeInfo.cs

```diff
@@ -68,6 +69,28 @@ internal static int InternalConvertToUtf32(String s, int index)
             }
             return ((int)s[index]);
         }
+
+        internal static int InternalConvertToUtf32(StringBuilder s, int index)
+        {
+            Debug.Assert(s != null, "s != null");
+            Debug.Assert(index >= 0 && index < s.Length, "index < s.Length");
+
+            if (index < s.Length - 1)
+            {
+                int temp1 = (int)s[index] - HIGH_SURROGATE_START;
```

Comment:
would there be any perf difference if you pulled out `s[index]` to a separate variable? (to avoid calling indexer twice)

Reply:
I see - this is a copy of the method above

Reply:
in general case, this s[index] would be used once.  so I prefer not introducing a local variable for that

Reply:
Is that the common case?  StringBuilder's indexer is potentially quite slow.

Reply:
We are going to use the StringBuilder index twice only in case of Surrogates which is not a common case.

Reply:
@tarekgh - it looks like on every index except last to me - might be worth to take a look at compiled code and see what it compiles it to (not necessarily in this PR though)